### PR TITLE
HPCC-29450 LDAP SecLib cannot retrieve perms of nested file scopes

### PR DESCRIPTION
**Pull request by RussWhitehead**
…e scopes for 389 DS

For OpenLDAP (389DS), only specify the deepest OU level when searching for OU security descriptors

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by wangkx — COMMENTED** [review]
@RussWhitehead the changes should work for 389DS. Are the code lines you are changing shared with other types of LDAP? Could the changes break (or downgrade) other types of LDAP?

**Comment by wangkx** [comment]
@RussWhitehead if that is the case, should you add the cn.appendf("ou=") to line 5398 (or move the code out of the if/else block)?

File: system/security/LdapSecurity/ldapconnection.cpp [hunk 1]
```diff
@@ -5399,7 +5399,11 @@ class CLdapClient : implements ILdapClient, public CInterface
                     }
                     else
                     {
-                        cn.append(curlen, curscope);
+                        if (cn.isEmpty())//only process leftmost part of ou, so "ou=s3,ou=s2,ou=s1" only specify ou=s3
+                        {
+                            cn.appendf("ou=");
```
Comment (wangkx) [inline system/security/LdapSecurity/ldapconnection.cpp:5404]: Better use the append()?

Comment (RussWhitehead) [reply system/security/LdapSecurity/ldapconnection.cpp:5404]: Done

**Review by wangkx — APPROVED** [review]
@RussWhitehead looks fine.

**Comment by RussWhitehead** [comment]
@ghalliday Please merge

**Comment by ghalliday** [comment]
@RussWhitehead should this target an earlier version (e.g. 9.0.x since it is a bug fix, and looks unlikely to break anything)

**Comment by RussWhitehead** [comment]
@ghalliday Rebased to 9.0.x

**Pull request merged**